prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -185,6 +185,9 @@ class ParserSpec extends AnyFunSpec with Matchers {
     parseError("foo{1}")
     parseError("{}")
     parseError("foo{__name__=\"bar\"}")
+    parseError("foo{__name__=\"foo\"}")
+    parseError("foo{_metric_=\"bar\"}")
+    parseError("foo{_metric_=\"foo\"}")
```

Comment:
Does something like `{_metric_="foo"}` still parse successfully? I'm not 100% sure if `metricName` accounts for both `foo{~}` and `{_metric_="foo"}`

Reply:
I'll add tests for those too but yes it parses. I'm not sure is ```{_metric_="foo"}``` is equivalent to ```foo{~}``` as ```_metric_``` is more internal to FiloDB

Reply:
Pushed more tests

Reply:
Ack, lgtm. Good point about `_metric_`-- I'm also not sure.

Reply:
`{__name__="foo", _metric_="foo"}` should fail.